os/pm/pm_dvfs.c

```diff
@@ -0,0 +1,110 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdint.h>
+#include <assert.h>
+#include <tinyara/pm/pm.h>
+
+#include "pm.h"
+
+#ifdef CONFIG_PM_DVFS
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_dvfs
+ *
+ * Description:
+ *   This internal function is called to reduce the frequency of operation
+ *   of the secondary AP core of AI_Dual chipset.
+ *
+ * Input Parameters:
+ *   div_lvl - voltage frequency scaling level
+ *
+ * Returned Value:
+ *   None.
+ *
+ * Assumptions:
+ *   Incorrect frequency scaling level given as input will lead to an assert.
+ *
+ ****************************************************************************/
+
+void pm_dvfs(int div_lvl)
+{
+	DEBUGASSERT(div_lvl >= 0 && div_lvl <= 3);
+	/* Only affects active mode power consumption
+	 * 0 -> 1.2GHz
+	 * 1 -> 600MHz
+	 * 2 -> 400Mhz
+	 * 3 -> 300Mhz
+	 */
+	/* Placed in os/arch/arm/src/amebasmart/amebasmart_pmc.c, can be invoked directly */
+	SOCPS_Tune_APFreq(div_lvl);
```

Comment:
This name is very specific to realtek, but this is a common file. ```up_set_dvfs``` or something similar so that other boards can also use it. @amogh-samsung could you please check this?

Reply:
This wrapper is provided by Amogh, if this API naming is too specific, I can just change it to up_set_dvfs(), is that okay?

Reply:
The wrapper is ok
Changing the name as you mentioned is OK from my side @edwakuwaku